public/js/components/Expressions.js

```diff
@@ -3,70 +3,134 @@ const { connect } = require("react-redux");
 const { bindActionCreators } = require("redux");
 const ImPropTypes = require("react-immutable-proptypes");
 // const classnames = require("classnames");
+const Svg = require("./utils/Svg");
 const actions = require("../actions");
 const { getExpressions, getPause } = require("../selectors");
-// const ObjectInspector = React.createFactory(require("./ObjectInspector"));
+const Rep = React.createFactory(require("./Rep"));
 // const { truncateStr } = require("../utils/utils");
 const { DOM: dom, PropTypes } = React;
 
 require("./Expressions.css");
 
 const Expressions = React.createClass({
   propTypes: {
-    pauseInfo: ImPropTypes.map,
     expressions: ImPropTypes.list,
     addExpression: PropTypes.func,
     updateExpression: PropTypes.func,
-    evaluateExpression: PropTypes.func,
-    loadObjectProperties: PropTypes.func,
-    command: PropTypes.func
+    deleteExpression: PropTypes.func
   },
 
   displayName: "Expressions",
 
-  addExpression(e) {
+  addExpression({ id }, e) {
     if (e.key === "Enter") {
+      const { addExpression } = this.props;
       const expression = {
         input: e.target.value
       };
-      if (e.target.id) {
-        expression.id = e.target.id.split("-").pop();
+      if (id !== undefined) {
+        expression.id = id;
       }
-      this.props.addExpression(expression);
+      e.target.value = "";
+      addExpression(expression);
     }
   },
 
-  updateExpression(e) {
+  updateExpression({ id }, e) {
+    e.stopPropagation();
+    const { updateExpression } = this.props;
     const expression = {
-      id: e.target.id.split("-").pop(),
-      input: e.target.textContent.split(" --> ")[0]
+      id,
+      input: e.target.textContent
     };
-    this.props.updateExpression(expression);
+    updateExpression(expression);
+  },
+
+  renderExpressionValue(value) {
+    if (!value) {
+      return;
+    }
+    if (value.exception) {
+      return Rep({ object: value.exception });
+    }
+    if (typeof value.result === "boolean") {
+      return `${value.result}`;
+    }
+    if (typeof value.result !== "object") {
+      return value.result;
+    }
+    if (typeof value.result === "object") {
+      return Rep({ object: value.result });
+    }
+  },
+
+  deleteExpression(expression, e) {
+    e.stopPropagation();
+    const { deleteExpression } = this.props;
+    deleteExpression(expression);
+  },
+
+  renderExpressionUpdating(expression) {
+    return dom.span(
+      { className: "expression-input-container" },
+      dom.input(
+        { type: "text",
+          className: "input-expression",
+          dataId: expression.id,
+          id: "expressionInputId-" + expression.id,
+          onKeyPress: this.addExpression.bind(this, expression),
```

Comment:
I think it's more idiomatic to do `(e) => this.addExpression(e, expression)`

Also, if this is a keypress handler, the function should be `this.inputKeyPress`